### PR TITLE
Fede+Fabo/Support ledger cosmos app v1.5 and older versions

### PR DESCRIPTION
**Pull request by fedekunze**
**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added entries in `PENDING.md` with issue # and GitHub username
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by fedekunze** [comment]
this PR needs to be updated to support the newest `1.5.0` app version once it's released

File: src/components/common/TmSessionHardware.vue [hunk 1]
```diff
@@ -45,26 +57,59 @@ export default {
...
-    connectionError: null
+    connectionError: null,
+    address: null
   }),
+  computed: {
+    submitCaption() {
+      return {
+        connect: "Sign In",
+        detect: "Waiting for Ledger",
+        confirmAddress: "Cofirming Address"
```
Comment (jbibla) [inline src/components/common/TmSessionHardware.vue]:
```suggestion
        confirmAddress: "Confirming Address"
```

**Comment by jbibla** [comment]
i don't totally understand this PR.

what is `confirmAddress`?

also, i am getting an `instruction not supported` error message when I sign in with ledger. and i don't see any way to confirm address or see the address when creating txs if i sign in with address.

**Comment by faboweb** [comment]
I forgot: You need to install the Ledger firmware 1.5.0 to test the new address feature: https://github.com/cosmos/ledger-cosmos/blob/master/docs/BUILD.md

> instruction not supported

I will investigate!

**Comment by faboweb** [comment]
Fixed compatibility with version `1.0.1`

**Comment by jbibla** [comment]
> You need to install the Ledger firmware 1.5.0 to test the new address feature: 

I would expect to see a warning that says my version is old and should be upgraded.

**Comment by faboweb** [comment]
The warning is commented out as the version is not yet available on Ledger Live.

File: src/components/common/TmSessionHardware.vue [hunk 2]
```diff
@@ -13,8 +13,16 @@
...
       </div>
       <div class="tm-session-main">
-        <HardwareState :loading="status === `detect` ? true : false">
-          Please plug in your Ledger&nbsp;Nano&nbsp;S and open the Cosmos app
+        <HardwareState :loading="status === `connect` ? false : true">
+          <template v-if="status === `connect` || status === `detect`">
+            Please plug in your Ledger&nbsp;Nano&nbsp;S and open the Cosmos app
+          </template>
+          <template v-if="status === `confirmAddress`">
+            Signin in with the address
```
Comment (jbibla) [inline src/components/common/TmSessionHardware.vue]:
```suggestion
            Sign in with this address
```

File: src/components/common/TmSessionHardware.vue [hunk 3]
```diff
@@ -13,8 +13,16 @@
...
+            <span class="address">{{ address }}</span
+            >.<br />
+            Please confirm on your Ledger.
```
Comment (jbibla) [inline src/components/common/TmSessionHardware.vue]: is there an action associated with this confirm message? if no, it should say "Please confirm that your address is correct" or something like that.

File: src/components/common/TmSessionHardware.vue [hunk 4]
```diff
@@ -45,26 +57,59 @@ export default {
...
-        await this.$store.dispatch(`connectLedgerApp`)
-      } catch (error) {
-        this.setStatus(`connect`)
-        this.setConnectionError(error.message)
+        await this.$store.dispatch("confirmLedgerAddress")
+        return true
+      } catch ({ message }) {
+        switch (message) {
+          case `Transaction rejected`:
+            this.connectionError = `Account address rejected`
```
Comment (jbibla) [inline src/components/common/TmSessionHardware.vue]: why would an address get rejected?

File: src/vuex/modules/ledger.js [hunk 5]
```diff
@@ -1,15 +1,69 @@
...
+      throw new Error(timeoutMessag)
+    case `Cosmos app does not seem to be open`:
+      throw new Error(`Cosmos app is not open`)
+    case `Command not allowed`:
+      throw new Error(`Transaction rejected`)
+    case `Transaction rejected`:
+      throw new Error(rejectionMessage)
+    case `Unknown error code`:
+      throw new Error(`Ledger's screensaver mode is on`)
+    case `Instruction no supported`:
```
Comment (jbibla) [inline src/vuex/modules/ledger.js]:
```suggestion
    case `Instruction not supported`:
```

File: src/vuex/modules/ledger.js [hunk 6]
```diff
@@ -1,15 +1,69 @@
...
+      throw new Error(
+        `Your Ledger Cosmos App is probably outdated. ` +
```
Comment (jbibla) [inline src/vuex/modules/ledger.js]:
```suggestion
        `Your Cosmos Ledger App is not up to date. ` +
```

File: src/vuex/modules/ledger.js [hunk 7]
```diff
@@ -1,15 +1,69 @@
...
+      throw new Error(error_message)
+  }
+}
+
+const checkAppMode = (chainId, response) => {
+  const { test_mode } = response
+
+  if (test_mode && chainId.startsWith(`cosmoshub`)) {
+    throw new Error(
+      `DANGER: The Cosmos Ledger app in test mode shouldn't be used on mainnet!`
```
Comment (jbibla) [inline src/vuex/modules/ledger.js]:
```suggestion
      `DANGER: The Cosmos Ledger app is in test mode and shouldn't be used on mainnet!`
```

File: src/vuex/modules/ledger.js [hunk 8]
```diff
@@ -82,61 +131,86 @@ export default () => {
...
-      } catch (error) {
-        Sentry.captureException(error)
-        commit(`setLedgerError`, error)
-        throw error
+    async connectLedgerApp({ commit, dispatch }) {
+      await dispatch(`pollLedgerDevice`)
+      await dispatch(`createLedgerAppInstance`)
+      const address = await dispatch(`getLedgerAddressAndPubKey`)
+      commit(`setLedgerConnection`, true)
+      // DEPRECATION enable once 1.5.0 is available on Ledger Live
```
Comment (jbibla) [inline src/vuex/modules/ledger.js]: 1.5 was released. still in dev mode.

File: src/vuex/modules/ledger.js [hunk 9]
```diff
@@ -82,61 +131,86 @@ export default () => {
...
+        // we can't check the address on an old cosmos app
+        return true
       }
+
+      const response = await state.cosmosApp.getAddressAndPubKey(
+        BECH32PREFIX,
+        HDPATH
+      )
+      checkLedgerErrors(state, response, {
+        rejectionMessage: "User rejected the displayed address"
```
Comment (jbibla) [inline src/vuex/modules/ledger.js]:
```suggestion
        rejectionMessage: "Displayed address was rejected"
```

File: src/components/common/TmSessionHardware.vue (hunk 3 above)
Comment (faboweb) [reply src/components/common/TmSessionHardware.vue]: Yes it is associated with an action on the Ledger. Please test these critical features on your Ledger if they come up.

File: src/components/common/TmSessionHardware.vue (hunk 4 above)
Comment (faboweb) [reply src/components/common/TmSessionHardware.vue]: The address approval is associated with an action, where you can "reject" the address.

File: src/vuex/modules/ledger.js (hunk 8 above)
Comment (faboweb) [reply src/vuex/modules/ledger.js]: Ok I enable this warning then.

**Pull request merged**